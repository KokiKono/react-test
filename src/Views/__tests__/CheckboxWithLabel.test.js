import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import CheckboxWithLabel from '../CheckboxWithLabel';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

test('CheckboxWithLabel changes the text after click', () => {
  // Render a checkbox with label in the document
  const checkbox = shallow(
    <CheckboxWithLabel labelOn="On" labelOff="Off" />
  );

  expect(checkbox.text()).toEqual('Off');
  expect(toJson(checkbox)).toMatchSnapshot();

  checkbox.find('input').simulate('change');

  expect(checkbox.text()).toEqual('On');
});

it('changes the label when click', () => {
  const wrapper = shallow(
    <CheckboxWithLabel labelOn="オン" labelOff="オフ" />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
  // onClick
  wrapper.find('input').simulate('change');
  expect(toJson(wrapper)).toMatchSnapshot();

})
