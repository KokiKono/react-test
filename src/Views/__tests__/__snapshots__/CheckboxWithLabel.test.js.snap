// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel changes the text after click 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
</label>
`;

exports[`changes the label when click 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  オフ
</label>
`;

exports[`changes the label when click 2`] = `
<label>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  オン
</label>
`;
